subprojects/core/src/integTest/groovy/org/gradle/internal/scan/config/BuildScanConfigIntegrationTest.groovy

```diff
@@ -36,9 +37,14 @@ Please see https://gradle.com/scans/help/gradle-incompatible-plugin-version for
                     def c = services.get(${BuildScanConfigProvider.name}).collect([getVersion: { "$pluginVersionNumber" }] as $BuildScanPluginMetadata.name) 
                     println "buildScan.enabled: " + c.enabled 
                     println "buildScan.disabled: " + c.disabled 
-                """
+                """.stripIndent()
             }
 
+            buildFile << """    
+                def pluginApplied = services.get(${BuildScanPluginApplied.name}).isBuildScanPluginApplied()
+                println "buildScan plugin applied: " + pluginApplied
+            """.stripIndent()
```

Comment:
Is there no other way to tell if the build scan plugin was applied? Like, check if we have it among the applied plugins or something? Adding a production type just for this seems unnecessarily complicated.

Reply:
I discussed with @ldaley about it. I had the check for the applied plugin before. I guess checking the flag is faster then checking if the plugin has been applied to the root project. It should also be more future proof, given that we have a problem in getting the right root project when running with composite builds or in the buildSrc project.